pkg/apis/apps/v1alpha1/uniteddeployment_types.go

```diff
@@ -0,0 +1,229 @@
+	// If unspecified, defaults to ReplicaSet
+	// +optional
+	SubsetType *metav1.TypeMeta `json:"subsetType,omitempty"`
+
+	// StatefulSet template
+	StatefulSet *StatefulSetTemplateSpec `json:"statefulSet,omitempty"`
+
+	// ReplicaSet template
+	ReplicaSet *ReplicaSetTemplateSpec `json:"replicaSet,omitempty"`
+}
+
+// StatefulSetTemplateSpec defines the subset template of StatefulSet
+type StatefulSetTemplateSpec struct {
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              appsv1.StatefulSetSpec `json:"spec,omitempty"`
+}
+
+// ReplicaSetTemplateSpec defines the subset template of ReplicasSet
+type ReplicaSetTemplateSpec struct {
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              appsv1.ReplicaSetSpec `json:"spec,omitempty"`
+}
+
+// UnitedDeploymentUpdateStrategy defines the update strategy of UnitedDeployment
+type UnitedDeploymentUpdateStrategy struct {
+	// Pause the rollout of all subset worloads
+	// +optional
+	Pause bool `json:"pause,omitempty"`
+
+	// Indicate the partition of each subset
+	Partitions map[string]int32 `json:"partitions,omitempty"`
+}
+
+// Topology defines the spread detail of each subset under UnitedDeployment
+type Topology struct {
+	// Contains the details of each subset
+	Subsets []Subset `json:"subsets,omitempty"`
+}
+
+// Subset defines the detail of a subset
+type Subset struct {
+	// Indicate the name of this subset, which will be used to generate
+	// subset workload name in the format '<deployment-name>-<subset-name>'
+	Name string `json:"name,omitempty"`
+
+	// Indicate the node select strategy to form the subset
+	NodeSelector corev1.NodeSelector `json:"nodeSelector,omitempty"`
+
+	// Indicate the number of pod replicas of this subset.
+	// If nil, the number of replicas in this subset is determined by controller.
+	// +optional
+	Replicas *intstr.IntOrString `json:"replicas,omitempty"`
+}
+
+// UnitedDeploymentStatus defines the observed state of UnitedDeployment
+type UnitedDeploymentStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	// observedGeneration is the most recent generation observed for this InPlaceSet. It corresponds to the
+	// InPlaceSet's generation, which is updated on mutation by the API Server.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// selector is a label query over pods that should match the replica count.
+	// It must match the template's labels.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
+	Selector *metav1.LabelSelector `json:"selector"`
+
+	// The number of ready replicas.
+	// +optional
+	ReadyReplicas int32 `json:"readyReplicas,omitempty"`
+
+	// Replicas is the most recently observed number of replicas.
+	Replicas int32 `json:"replicas,omitempty"`
+
+	// The number of pods in current version
```

Comment:
across all subsets?

Reply:
I think if it is necessary in the future, we can add a map to show the spread detail of updated pod.